guppylang/definition/pytket_circuits.py

```diff
@@ -210,9 +210,21 @@ def compile_outer(
                     )
 
                 # Symbolic parameters get passed at the end.
-                param_wires = list(
-                    outer_func.inputs()[len(self.input_circuit.q_registers) :]
-                )
+                # We assume they are given in lexicographic order by the user.
+                # We need to sort them according to the metadata order.
+                # TODO: Allow to be passed as array if `use_arrays=True`.
+                param_wires = []
+                if "TKET1.input_parameters" in hugr_func.metadata:
```

Comment:
We should probably error out if the circuit has parameters but this metadata is missing?